server/app/views/HtmlBundle.java

```diff
@@ -26,30 +35,30 @@ public class HtmlBundle {
   private String language = "en";
 
   private ArrayList<String> bodyStyles = new ArrayList<>();
-  private ArrayList<Tag> footerContent = new ArrayList<>();
-  private ArrayList<Tag> footerScripts = new ArrayList<>();
+  private ArrayList<DivTag> footerContent = new ArrayList<>();
+  private ArrayList<ScriptTag> footerScripts = new ArrayList<>();
   private ArrayList<String> footerStyles = new ArrayList<>();
-  private ArrayList<Tag> headScripts = new ArrayList<>();
-  private ArrayList<Tag> headerContent = new ArrayList<>();
+  private ArrayList<ScriptTag> headScripts = new ArrayList<>();
+  private ArrayList<NavTag> headerContent = new ArrayList<>();
   private ArrayList<String> headerStyles = new ArrayList<>();
   private ArrayList<Tag> mainContent = new ArrayList<>();
   private ArrayList<String> mainStyles = new ArrayList<>();
-  private ArrayList<EmptyTag> metadata = new ArrayList<>();
+  private ArrayList<MetaTag> metadata = new ArrayList<>();
   private ArrayList<Modal> modals = new ArrayList<>();
-  private ArrayList<Tag> stylesheets = new ArrayList<>();
+  private ArrayList<LinkTag> stylesheets = new ArrayList<>();
   private ArrayList<ToastMessage> toastMessages = new ArrayList<>();
 
   public HtmlBundle addBodyStyles(String... styles) {
     bodyStyles.addAll(Arrays.asList(styles));
     return this;
   }
 
-  public HtmlBundle addFooterContent(Tag... tags) {
+  public HtmlBundle addFooterContent(DivTag... tags) {
```

Comment:
Is there any specific reason that we need to specify DivTags here? This feels like a case where we want to use a more general type rather than something specific.

Reply:
Ok, but what about the `footerScripts` and `headScripts` fields? Should those be left as `ScriptTag`?

Reply:
I'd leave those as `ScriptTags`, yes. I think the general issue I was pointing out was that the method specifies content, which doesn't imply a specific tag type to me. In those cases, I think a generalized type (e.g. `Tag`) makes sense. In cases where callers would assume/need a specific type, it makes sense to return a more specific type.

There's no hard or fast rule here. One potential indicator that a more general type is needed is if we end up modifying the actual DOM structure returned just to satisfy the Java type system.